### PR TITLE
fix: fix allocateIdleResource bug

### PR DESCRIPTION
**Pull request by RamezesDong**
My implementation of bug #2624

### CODE REVIEW

File: pkg/scheduler/api/node_info.go [hunk 1]
```diff
@@ -411,6 +412,17 @@ func (ni *NodeInfo) allocateIdleResource(ti *TaskInfo) error {
 		ni.Idle.Sub(ti.Resreq)
 		return nil
 	}
+	// if resReq larger than node idle, set node idle to zero
```
[the remark this answers is not in the document]
Comment (Thor-wl) [reply pkg/scheduler/api/node_info.go:415]: > When the requested resource is greater than the idle resource of the node, no resource is allocated and the idle resource does not change. Why does the resource amount of the req need to be deducted?

+1, @RamezesDong Can you give more interpretations about that?

[the remark this answers is not in the document]
Comment (RamezesDong) [reply pkg/scheduler/api/node_info.go:415]: In the scenario I encountered, there is an inference running in the node with 4 GPUs. A GPU became unhealthy, and the requested resource (4 GPUs) is greater than the idle resource of the node (3 GPUs). Another pod was scheduled to this node, then an error occurred.
```
Allocate failed due to requested number of devices unavailable for nvidia.com/gpu. Requested: 4, Available: 3, which is unexpected
```
The volcano we maintained has merged this pr, and this problem rarely happens. You can hold this PR first, and we will find other possible reasons.

**Comment by hwdef** [comment]
@RamezesDong Why close this?

**Pull request closed without merging**